frame/support/procedural/src/pallet/expand/store_trait.rs

```diff
@@ -0,0 +1,55 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::pallet::Def;
+use syn::spanned::Spanned;
+
+/// If attribute `#[pallet::generate_store(..)]` is defined then:
```

Comment:
hmmm it seems like this is optional? I think it should be explained in the respective doc why and how it should be used.

Reply:
(I know there has been some historical debate about this traits, but tbh I don't know either what purpose it serves and why it is controversial.)

Reply:
I agree, (no I didn't underdocumented because of my aversion to it :0 )